docs/usage/configuration.md

```diff
@@ -104,6 +104,36 @@ The [Git tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) format used by
 
 **Note**: The `tagFormat` must contain the `version` variable exactly once and compile to a [valid Git reference](https://git-scm.com/docs/git-check-ref-format#_description).
 
+### tagAnnotate
+
+Type: `Boolean`<br>
+Default: `false`<br>
+CLI arguments: `--ta`, `--tag-annotate`
+
+Whether to create an unsigned, annotated tag. The default is `false` which will create an lightweight tag.
+
+**Note**: `tagSign` takes precedence over `tagAnnotate`. The git config `tag.gpgsign` takes precedence over both.
+
+### tagSign
+
+Type: `Boolean`<br>
+Default: `false`<br>
+CLI arguments: `--ts`, `--tag-sign`
+
+Whether to create a GPG-signed tag, using the configured GPG signing key. Signed tags must have a message.
+
+**Note**: `tagSign` takes precedence over `tagAnnotate`. The git config `tag.gpgsign` takes precedence over both.
+
+### tagMessage
+
+Type: `String`<br>
+Default: `release ${nextRelease.version}`<br>
+CLI arguments: `-tm`, `--tag-message`
+
+The message for the tag. The tag message is generated with [Lodash template](https://lodash.com/docs#template) and will be compiled with the `nextRelease` variable.
```

Comment:
`nextRelease` being an internal variable, there should be a link to [its documentation](https://semantic-release.gitbook.io/semantic-release/developer-guide/js-api#nextrelease).